exporters/otlp/otlptrace/otlptracegrpc/client.go

```diff
@@ -196,9 +198,12 @@ func (c *client) UploadTraces(ctx context.Context, protoSpans []*tracepb.Resourc
 	defer cancel()
 
 	return c.requestFunc(ctx, func(iCtx context.Context) error {
-		_, err := c.tsc.Export(iCtx, &coltracepb.ExportTraceServiceRequest{
+		resp, err := c.tsc.Export(iCtx, &coltracepb.ExportTraceServiceRequest{
 			ResourceSpans: protoSpans,
 		})
+		if resp != nil && resp.PartialSuccess != nil {
```

Comment:
Ditto.

Reply:
See https://go.dev/ref/spec#Logical_operators, "The right operand is evaluated conditionally." This is sometimes called "short-circuit logical evaluation", it's common in all programming languages that I'm familiar with.